"""changed character limits on constraints model

Revision ID: 881027dde40a
Revises: ffdc0a98111c
Create Date: 2021-06-03 17:40:51.111307

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '881027dde40a'
down_revision = 'ffdc0a98111c'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('generators',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('title', sa.String(length=40), nullable=False),
    sa.Column('seed', sa.Integer(), nullable=False),
    sa.Column('iteration', sa.Integer(), nullable=True),
    sa.Column('user_id', sa.Integer(), nullable=False),
    sa.ForeignKeyConstraint(['user_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('constraints',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('generator_id', sa.Integer(), nullable=False),
    sa.Column('uppercase_letters', sa.String(length=260), nullable=False),
    sa.Column('lowercase_letters', sa.String(length=260), nullable=False),
    sa.Column('numbers', sa.String(length=100), nullable=False),
    sa.Column('symbols', sa.String(length=330), nullable=False),
    sa.Column('pw_length', sa.Integer(), nullable=False),
    sa.Column('required_uppercase', sa.Integer(), nullable=False),
    sa.Column('required_numbers', sa.Integer(), nullable=False),
    sa.Column('required_symbols', sa.Integer(), nullable=False),
    sa.Column('allow_duplicates', sa.Boolean(), nullable=False),
    sa.ForeignKeyConstraint(['generator_id'], ['generators.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('constraints')
    op.drop_table('generators')
    # ### end Alembic commands ###
